File: leetcode/078-subsets.py
class Solution(object):
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        ret = [[]]
        for n in nums:
            ret = ret + [x + [n] for x in ret]
        return ret;

File: leetcode/078-subsets.cpp
class Solution
{
 public:
  vector<vector<int>> subsets(vector<int>& nums)
  {
    vector<vector<int>> ret = {{}};
    for (int i = 0; i < nums.size(); ++i) {
      for (int j = 0, n = ret.size(); j < n; ++j) {
        vector<int> v = ret[j];
        v.push_back(nums[i]);
        ret.push_back(v);
      }
    }
    return ret;
  }
};
